pyscf_ipu/experimental/numerics.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+from functools import wraps
+
+import jax.numpy as jnp
+import numpy as np
+
+
+def apply_fpcast(v, dtype):
+    if isinstance(v, jnp.ndarray) and v.dtype.kind == "f":
```

Comment:
Does `np.issubdtype(d, np.floating)` work here?
I am wondering if JAX is using numpy dtyps underneath

Reply:
Nice, that does work.  Will update this here